src/main/webapp/app/exercises/text/manage/example-text-submission/example-text-submission.component.ts

```diff
@@ -260,25 +260,44 @@ export class ExampleTextSubmissionComponent extends TextAssessmentBaseComponent
             return;
         }
         this.tutorParticipationService.assessExampleSubmission(this.exampleSubmissionForNetwork(), this.exerciseId).subscribe(
-            () => this.jhiAlertService.success('artemisApp.exampleSubmission.assessScore.success'),
+            () => {
+                this.markAllFeedbackToCorrect();
+                this.jhiAlertService.success('artemisApp.exampleSubmission.assessScore.success');
+            },
             (error: HttpErrorResponse) => {
                 const errorType = error.headers.get('x-artemisapp-error');
 
-                switch (errorType) {
-                    case 'error.tooLow':
-                        this.jhiAlertService.error('artemisApp.exampleSubmission.assessScore.tooLow');
-                        break;
-                    case 'error.tooHigh':
-                        this.jhiAlertService.error('artemisApp.exampleSubmission.assessScore.tooHigh');
-                        break;
-                    default:
-                        onError(this.jhiAlertService, error);
-                        break;
+                if (errorType === 'error.invalid_assessment') {
+                    this.markAllFeedbackToCorrect();
+
+                    // Mark all wrongly made feedbacks accordingly.
+                    const correctionErrors: FeedbackCorrectionError[] = JSON.parse(error['error']['title'])['errors'];
+                    correctionErrors.forEach((res) => {
+                        const textBlockRef = this.textBlockRefs.find((ref) => ref.feedback?.reference === res.reference);
+                        if (textBlockRef != undefined && textBlockRef.feedback != undefined) {
+                            textBlockRef.feedback.correctionStatus = res.type;
+                        }
+                    });
+
+                    const errorMessage =
+                        correctionErrors.length === 0 ? 'artemisApp.exampleSubmission.submissionValidation.missing' : 'artemisApp.exampleSubmission.submissionValidation.wrong';
+                    this.jhiAlertService.error(errorMessage, { mistakeCount: correctionErrors.length });
+                } else {
+                    onError(this.jhiAlertService, error);
```

Comment:
I am wondering if you could not factor out the logic for marking all wrongly made feedback as well as marking it to correct (below) into a method that can be used by `ExampleTextSubmissionComponent` and ExampleModelingSubmissionComponent (and probably some more in the future?). I saw, that it's not exactly the same, but maybe you can prepare the textBlockRefs before passing it into a method, so that it's actually the same as for the `ExampleModelingSubmissionComponent`. Just an idea.

Reply:
I removed the duplication an extracted it into a separate component.